utils.go

```diff
@@ -430,7 +430,8 @@ func setInterfaceMacAddress(ifName, mac string) error {
 
 func findMACAddressForContainer(containerID, rancherID string) (string, error) {
 	var mf macfinder.MACFinder
-	mf, err := metadata.NewMACFinderFromMetadata()
+	metadataAddress := os.Getenv("RANCHER_METADATA_ADDRESS")
+	mf, err := metadata.NewMACFinderFromMetadata(metadataAddress)
```

Comment:
Pre-existing to this PR, so no need to change in this PR but creating a new metadata client every time we need to find a mac address seems wasteful. Considering reuisng a client in a future PR.